### DISCUSSION TITLE
whethcer can open the old c++ source

### DISCUSSION THREAD

**Discussion by userMocius** [risingwavelabs/risingwave#2821] [General]
so we can learn it

**Comment by yingjunwu** [comment]
Thanks for your interest in our project. Unfortunately we do not have plan to open-source our deprecated C++ code. The main reasons are: 1) it is not compatible with the testing frameworks or tool chains we are currently using; 2) open-sourcing the old code still requires us to do plenty of cleanup works, such as adding licenses, fixing bugs, editing comments, etc., which can be a distraction; 3) we don't really have enough bandwidth to maintain the C++ version if it's open sourced.

While we moved from C++ to Rust, the design principles of RisingWave didn't really change at all. So you may still refer to the latest doc to learn how RisingWave works.

Thank you again for your interest!
